src/hooks/data/useBetsSummaryBySelection.ts

```diff
@@ -0,0 +1,119 @@
+import { useMemo } from 'react'
+import { useQuery } from '@apollo/client'
+import { type Address, formatUnits, parseUnits } from 'viem'
+
+import { useApolloClients } from '../../contexts/apollo'
+import { GameStatus } from '../../utils/getGameStatus'
+import { GameBetsDocument, type GameBetsQuery, type GameBetsQueryVariables } from '../../docs/prematch/gameBets'
+import { BetResult } from '../../docs/prematch/types'
+import { useChain } from '../../contexts/chain'
+import { ODDS_DECIMALS } from '../../config'
+
+
+export type BetsSummaryBySelection = Record<string, string>
+
+type Props = {
+  account: Address
+  gameId: string
+  gameStatus: GameStatus
+  keyStruct?: 'outcomeId' | 'conditionId-outcomeId'
+}
+
+const DIVIDER = 18
+
+export const useBetsSummaryBySelection = ({ account, gameId, gameStatus, keyStruct = 'outcomeId' }: Props) => {
+  const { prematchClient } = useApolloClients()
+  const { betToken } = useChain()
+
+  const variables = useMemo<GameBetsQueryVariables>(() => ({
+    actor: account?.toLowerCase(),
+    gameId,
+  }), [ account, gameId ])
+
+  const { data, loading, error } = useQuery<GameBetsQuery, GameBetsQueryVariables>(GameBetsDocument, {
+    variables,
+    ssr: false,
+    client: prematchClient!,
+    skip: !account || gameStatus !== GameStatus.Resolved,
+  })
+
+  const { bets: prematchBets, liveBets } = data || {}
+
+  const betsSummary = useMemo<BetsSummaryBySelection>(() => {
+    if (!prematchBets?.length && !liveBets?.length) {
+      return {}
+    }
+
+    const rawOne = parseUnits('1', ODDS_DECIMALS)
+
+    const rawSummary = [ ...(prematchBets || []), ...(liveBets || []) ].reduce<Record<string, bigint>>((acc, bet) => {
+      const { rawAmount: _rawAmount, rawPotentialPayout: _rawPotentialPayout, result, selections } = bet
+      const { freebet } = bet as GameBetsQuery['bets'][0]
+
+      const isExpress = selections.length > 1
+      const isWin = result === BetResult.Won
+
+      const rawAmount = BigInt(_rawAmount)
+      const rawBetDiff = freebet ? rawAmount : 0n
+      const rawPayout = BigInt(_rawPotentialPayout) - rawBetDiff
+
+      let rawOddsSummary = 0n
+
+      if (isExpress) {
+        selections.forEach(selection => {
+          const { rawOdds } = selection as GameBetsQuery['bets'][0]['selections'][0]
+
+          rawOddsSummary += BigInt(rawOdds) - rawOne
```

Comment:
let's remove this extra pass by moving it down to https://github.com/Azuro-protocol/sdk/pull/36/files#diff-9f5403c30befe89e52ff185c8ecf3440514c279844bf52d6c10d080b08c86726R73-R75

Reply:
we can't, it'll initiate loop for every selection in express, and we also need get `rawOddsSummary` before other calculation